pkg/kube/client.go

```diff
@@ -479,17 +479,23 @@ func (c *client) AllDiscoveryDo(ctx context.Context, istiodNamespace, path strin
 	if len(istiods) == 0 {
 		return nil, errors.New("unable to find any Istiod instances")
 	}
+	var errs error
 	result := map[string][]byte{}
 	for _, istiod := range istiods {
 		res, err := c.proxyGet(istiod.Name, istiod.Namespace, path, 15014).DoRaw(ctx)
 		if err != nil {
-			return nil, err
+			err = multierror.Append(errs, err)
+			continue
 		}
 		if len(res) > 0 {
 			result[istiod.Name] = res
 		}
 	}
-	return result, err
+	// If any Discovery servers responded, treat as a success
```

Comment:
should we give an warning or something still? might be confusing otherwise

Reply:
fixed